core/sdk-core/src/main/java/software/amazon/awssdk/core/SdkSystemSetting.java

```diff
@@ -90,6 +90,16 @@ public enum SdkSystemSetting implements SystemSetting {
 
     AWS_EC2_METADATA_SERVICE_ENDPOINT_MODE("aws.ec2MetadataServiceEndpointMode", "IPv4"),
 
+    /**
+     * The number of seconds (either as an integer or double) before a connection to the instance
+     * metadata service should time out. This value is applied to both the socket connect and read timeouts.
+     *
+     * The timeout can be configured using the system property "aws.ec2MetadataServiceTimeout". If not set,
+     * a default timeout is used. This setting is crucial for ensuring timely responses from the instance
+     * metadata service in environments with varying network conditions.
+     */
+    AWS_METADATA_SERVICE_TIMEOUT("aws.ec2MetadataServiceTimeout", "1"),
```

Comment:
Should we also add profile config? https://docs.aws.amazon.com/sdkref/latest/guide/feature-ec2-instance-metadata.html

Reply:
done